docs/internet-identity-spec.adoc

```diff
@@ -536,6 +539,23 @@ This flow is the boring default
 +
 (This could include a button to go to the management screen, or maybe this _is_ the management screen with a info box.)
 
+=== Flow: Setup recovery
+1. The user is offered two options for recovery
+  * A security key
+  * A BIP-39 seed phrase
+  * (Skip?)
+2. Depending on their choice
+ a) If they choose the seed phrase it is generated and displayed to them with a copy button and an explanation of having to keep this phrase secure
+ b) 👆 If they choose the security key they generate new credentials for the touched device
+3. The device is added to the users account with the `is_recovery` flag set
+
+=== Flow: Use recovery
+1. On the login page the user selects "Recover my account"
+2. The user is prompted for their user number
+3. If no `DeviceData with` `is_recovery` is found for the user number an error is displayed
+4. The user is asked to provide the seed phrase or 👆 their recovery security key
```

Comment:
I'm not sure what to do here when a user has configured multiple recovery devices yet.

Reply:
Presumably they’d have names, and the frontend would ask which one they want to use?

Reply:
If they had configured multiple security keys we could do the same thing we do on login right now, but you're right picking from a list of names might be the simplest for now.

Reply:
> If they had configured multiple security keys we could do the same thing we do on login right now

No, I don’t think so: Once you have non-WebAuthn keys (e.g. recovery codes), this will not work anymore: You can’t present those to the browser and make it do the right thing anymore.

Reply:
We can tell recovery codes from Webauthn devices by whether or not they have a CredentialId. That's how I'm planning to prompt the user to ☝️ or insert their seed phrase without them having to tell me which of the two options they chose for recovery.

Reply:
That might be short-sighted: If I use dfx to add a key, it will not have a CredentialId, but it also wouldn't be a recovery code. Better keep that information explicitly in the backend if you need to use it.

Reply:
Right... so an extra enum: `{ #platformAuth; #crossPlatformAuth; #recoveryCode; }` and that would keep it extensible and backwards compatible? Or would you make the full `DeviceData` a variant?

Reply:
Yes, extra enum, wrapped in `opt` so that we can decode existing fields that don't have it yet.

And I would maybe distinguish between form and purpose, i.e.
```
type = { #platformAuth, #crossPlatformAuth; #bipCodes; #plain } // plain for, e.g., dfx or other external tools
purpuse = { #hot; #recovery }
```
Because I may want to use a `crossPlatformAuth` for recovery.